ARCHITECTURE.md

```
@@ -0,0 +1,903 @@
+
+### Why Correlation IDs in Logs?
+
+**Problem**: With parallel execution, logs from different tests interleave. Hard to trace which log belongs to which test.
+
+**Solution**: Every TestOperation/SuiteOperation has a `correlationID: UUID`:
+```swift
+Logger.shared.info("Test started: \(testID)", correlationID: correlationID)
+```
+
+**Output**:
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)